src/main/java/de/retest/recheck/configuration/MavenProjectRootFinder.java

```diff
@@ -0,0 +1,54 @@
+package de.retest.recheck.configuration;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MavenProjectRootFinder implements ProjectRootFinder {
+
+	private static final Logger logger = LoggerFactory.getLogger( ProjectRootFinder.class );
+	private final Set<Path> indicator = new HashSet<>();
+
+	public MavenProjectRootFinder() {
+		indicator.add( Paths.get( "src/main/java" ) );
+		indicator.add( Paths.get( "src/test/java" ) );
+	}
+
+	@Override
+	public Optional<Path> findProjectRoot() {
+		return findProjectRoot( Paths.get( "" ).toAbsolutePath() );
+	}
+
+	@Override
+	public Optional<Path> findProjectRoot( final Path basePath ) {
```

Comment:
What would be an example for `basePath`?

Reply:
The path to a report.